### PR TITLE
[CBRD-25179] Add sql testcase for A core dump occurs when the CTE query is a false query

### PR DESCRIPTION
**Pull request by ssihil**
refer to http://jira.cubrid.org/browse/CBRD-25179

### CODE REVIEW

File: sql/_13_issues/_24_1h/cases/cbrd_25179.sql [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+--@queryplan
+WITH cte AS (SELECT col_a, col_b FROM   tbl_a  WHERE col_a LIKE Concat ('KOR-', NULL, '%'))
+SELECT /*+ recompile */ 1 j, tbl_b.* FROM  tbl_b INNER JOIN cte ON tbl_b.col_a = cte.col_a LEFT JOIN tbl_a ON tbl_a.col_b = cte.col_b;
+
+-- Query plan not created.
+--@queryplan
+SELECT /*+ recompile */ 1 k, a1.* FROM   ( WITH cte AS (SELECT col_a, col_b FROM   tbl_a  WHERE col_a LIKE Concat ('KOR-', NULL, '%'))
+SELECT * FROM  cte) a1 INNER JOIN ( WITH cte AS (SELECT col_a, col_b FROM   tbl_b  WHERE col_a LIKE Concat ('KOR-', NULL, '%'))
+SELECT * FROM  cte) a2 ON a1.col_a = a2.col_a LEFT JOIN tbl_b ON tbl_b.col_b = a2.col_b ;
+
```
Comment (kwonhoil) [inline sql/_13_issues/_24_1h/cases/cbrd_25179.sql:62]: Please add the query below.

--@queryplan
SELECT /*+ recompile */ 1 a, T.* FROM   ( WITH cte AS (SELECT col_a, col_b FROM   tbl_a  WHERE 1=0)
SELECT tbl_a.* FROM  tbl_a INNER JOIN cte ON tbl_a.col_a = cte.col_a LEFT JOIN tbl_b ON tbl_a.col_b = tbl_b.col_b ) t;


--@queryplan
SELECT /*+ recompile */ 1 a, T.* FROM   ( WITH cte AS (SELECT col_a, col_b FROM   tbl_a  WHERE col_a > 10000 and a <100 )
SELECT tbl_a.* FROM  tbl_a INNER JOIN cte ON tbl_a.col_a = cte.col_a LEFT JOIN tbl_b ON tbl_a.col_b = tbl_b.col_b ) t;



--@queryplan
SELECT /*+ recompile */ 1 a, T.* FROM   ( WITH cte AS (SELECT col_a, col_b FROM   tbl_a  WHERE col_b = '2024-13-01' )
SELECT tbl_a.* FROM  tbl_a INNER JOIN cte ON tbl_a.col_a = cte.col_a LEFT JOIN tbl_b ON tbl_a.col_b = tbl_b.col_b ) t;


--@queryplan
SELECT /*+ recompile */ 1 a, T.* FROM   ( WITH cte AS (select col_a, count(*) cnt from tbl_a group by col_b )
SELECT * FROM  cte  ) t;


--@queryplan
SELECT /*+ recompile */ 1 a, T.* FROM   ( WITH cte AS (select col_a, count(*) col_b from tbl_a group by col_b )
SELECT tbl_a.* FROM  tbl_a INNER JOIN cte ON tbl_a.col_a = cte.col_a LEFT JOIN tbl_b ON tbl_a.col_b = tbl_b.col_b  ) t;

Comment (ssihil) [reply sql/_13_issues/_24_1h/cases/cbrd_25179.sql:62]: reflected the review. thanks

Comment (bagus-kim) [reply sql/_13_issues/_24_1h/cases/cbrd_25179.sql:62]: add TestCase : Result (All Sucess) 
Please. Check


----------

drop table if exists t1, t2;

create table t1(c1 varchar(10), c2 varchar(10));
create table t2(c1 varchar(10), c2 varchar(10));


insert into t1 values('KOR-',NULL);
insert into t1 values('AME-',NULL);
insert into t2 values('KOR-','CUBRID')

--TC001 
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%')) 
	SELECT t2.* 
	FROM  t2 INNER JOIN cte ON t2.c1 = cte.c1 
) T;

--TC002
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%')) 
	SELECT t2.* 
	FROM  t2 INNER JOIN cte ON t2.c2 = cte.c2
) T;

--TC003
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC004
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%')) 
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c1 = cte.c1 
) T;

--TC005
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%')) 
	SELECT t1.* 
	FROM  t1 CROSS JOIN cte
) T;

--TC006
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1, c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%')) 
	SELECT t1.* 
	FROM  t1 CROSS JOIN cte
) T;

--TC007
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c1 as c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC008
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 LIKE Concat ('KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC009
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 = 'KOR-'||NULL||'%')
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC010
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 = NULL||'KOR-'||'adfqㅇㅋㅍㅊ')
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC011
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 = NULL)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC012
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 <> NULL)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC013
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 != NULL)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC014
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 is NULL)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC015
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  1 = 1)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC016
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  1 = 0)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC017
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  1 != 0)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC018
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  0 <> 1)
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC019
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM   t1 WHERE  c1 = Concat ('KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC020
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM t1 WHERE  c1 like CONCAT_WS(' ', 'KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;

--TC021
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM t1 WHERE  c1 like CONCAT_WS(NULL, 'KOR-', NULL, '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;
--TC022
SELECT /*+ recompile */ T.* 
FROM   (WITH cte AS (SELECT c2 FROM t1 WHERE  c1 like CONCAT_WS(NULL, NULL, 'KOR-', '%'))
	SELECT t1.* 
	FROM  t1 LEFT JOIN cte ON t1.c2 = cte.c2 
) T;
--TC023
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select NULL ) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC024
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select 1=0 ) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC025
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  bit_length(NULL) ) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC026
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = bit_length(NULL) ) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC026
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = FIND_IN_SET(NULL,'a,b,c,d')) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC027
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = lower(NULL)) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC028
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = left(NULL,3)) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC029
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = lpad(NULL, 10, '?')) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;
--TC030
SELECT /*+ recompile */ T.* 
FROM   (with cte as ( select  c2 from t1 where c2 = ltrim(NULL)) 
	SELECT t1.* 
	FROM  t1 JOIN cte
) T;

**Review by bagus-kim — COMMENTED** [review]
Add testcase. Please check

**Review by beyondykk9 — APPROVED** [review]

File: sql/_13_issues/_24_1h/cases/cbrd_25179.sql (hunk 1 above)
Comment (ssihil) [reply sql/_13_issues/_24_1h/cases/cbrd_25179.sql:62]: Thank you for the review.
It was applied except for similar types.

**Review by kwonhoil — APPROVED** [review]

**Review by tw-kang — APPROVED** [review]

**Review by bagus-kim — APPROVED** [review]

**Pull request merged**